junitparser/junitparser.py

```diff
@@ -388,6 +391,36 @@ def testsuites(self):
     def write(self, filepath=None, pretty=False):
         write_xml(self, filepath=filepath, pretty=pretty)
 
+    def group_by(self, key=None):
+        """
+        Split a TestSuite into multiple TestSuite using a TestCase attribute. For exemple, if I have a TestSuite like
+        this:
+          TestSuite(TestCase(classname=class1), TestCase(classname=class2))
+
+        and I split it using this method like this:
+          test_suite.group_by('classname')
+
+        I'll get this dict in return:
+        {'class1': TestSuite(TestCase(classname='class1')), 'class2': TestSuite(TestCase(classname='class2'))}
+
+        :param key: A TestCase attribute or a callable that returns an hashable python object
+        :return: A dict of key: TestSuites
+        """
+        if isinstance(key, six.string_types):
```

Comment:
Six is a good package, but if this is the only instance where it's been used, I don't think adding the dependency is worth it.

Reply:
That's a good point, plus str should always do the trick, even in python 2.